app/models/user.rb

```diff
@@ -0,0 +1,28 @@
+class User < ApplicationRecord
+  include BCrypt
+  has_many :comments
+
+  has_many :suggestions
+
+  validates :username, :email, :password, presence: true
+  validates :username, uniqueness: true
+
+
+  def password
+    @password ||= BCrypt::Password.new(password_hash)
+  end
+
+  def password=(new_password)
+    @password = BCrypt::Password.create(new_password)
+    self.password_hash = @password
+  end
+
+  def self.authenticate(username)
+    return nil unless user = find_by(username: username)
+    return user if user.password == plaintext_password
```

Comment:
Heads up: `plaintext_password` is not defined in this method. (It needs to be passed into the method.)